### ISSUE TITLE
Rule landmark-one-main is not filtered via tags 

### ISSUE THREAD

**Issue by felixzapata** [dequelabs/axe-core#785]
Hi, if you try to filter ignoring the `best-practice` rules in the results (for example, `['wcag2aa', 'wcag2a', 'section508']`), the rule `landmark-one-main` always appears.

This happen after the version 2.0.6. The only way to avoid this rule in the results is to disable it:

`'landmark-one-main': { enabled: false }`

**Comment by WilcoFiers** [comment]
I don't seem to be able to reproduce this. Are you using the ronOnly option correctly?

```js
  axe.run({ runOnly: {
    type: 'tag', values: ['wcag2a', 'wcag2aa']
  }}, function (err, res) {
    console.log(res.violations);
  });
```

**Comment by felixzapata** [comment]
No I'm not. Really I am using through `axe-webdriverjs` with the method `withRules`:

```
AxeBuilder(driver).withRules(['html-lang', 'image-alt']);
```

Really I do not know if the issue is in this repository or in the other one.

**Comment by WilcoFiers** [comment]
So are you doing this?
```js
AxeBuilder(driver).withRules(['wcag2a', 'wcag2aa']);
```

Because that shouldn't work, those are tags, not rules. You need the `.withTags()` method.

**Comment by felixzapata** [comment]
Sorry my mistake, in my previous comment  i wanted to say `withTags` instead of `withRules`.

**Comment by marcysutton** [comment]
I can confirm that best practice rules without WCAG tags are coming up with the `withTags` API method in axe-webdriverjs. But we should open it against that project, since it includes some custom code before calling `runOnly`. I can't reproduce it with `runOnly`.

**Comment by marcysutton** [comment]
Actually, I messed up in my testing. I can't reproduce it with this call to axe-webdriverjs:
```
AxeBuilder(browser)
    .withTags(['wcag2a', 'wcag2aa'])
    .analyze(function(results) {
        console.log(results.violations);
    });
```

For this page, I get 7 violations with that filter and 10 without (including the landmark rules). http://www.thebikerevolution.com/where-to-ride/road-bike-rides/bayshore-bikeway/

**Comment by felixzapata** [comment]
Ok, I will check it again to see why calling:

```javascript
AxeBuilder(browser)
    .withTags(['wcag2a', 'wcag2aa'])
```

the results shows me the best practice rule as a violation instead of filter it.
